### PR TITLE
Add info about storage transactions

### PR DESCRIPTION
**Pull request by ynewmann**

### CODE REVIEW

**Review by Cempl — APPROVED** [review]

File: docs/built-in-features/storage.md [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+### Replicator
+
+To provide the Storage, the Replicator should release [ReplicatorOnboardingTransaction](#replicatoronboardingtransaction) and prove its space with collateral. The real free space on the Replicator’s hard Drive MUST be at least twice larger than the mentioned in the [ReplicatorOnboardingTransaction](#replicatoronboardingtransaction) to provide the possibility of rollbacks.
+
+With a special algorithm, the Replicator will be automatically attached to a Drive.
+
+The replicator leaves the Drive if it was closed or the next `Billing Period` has not been paid. Also, the replicator could be excluded from the Drive based on the results of the verification.
+
+To offboard the Replicator should release the [ReplicatorOffBoardingTransaction](#replicatoroffboardingtransaction). The Replicator receives the payments for the work at the end of the `Billing Period`, and returns the deposit for the Drive, but losses part of its Streaming Deposit. This amount compensates for further data uploading during substitution.
```
Comment (wernling-proximax) [inline docs/built-in-features/storage.md]:
```suggestion
To offboard Storage, the Replicator should release [ReplicatorOffBoardingTransaction](#replicatoroffboardingtransaction). The Replicator receives the payments for the work at the end of the `Billing Period`, and returns the deposit for the Drive, but loses part of its Streaming deposit. This amount will be compensated for further data uploading during substitution.
```

File: docs/built-in-features/storage.md [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+
+To perform a modification, the Owner must form the action list that contains all actions which must be executed by the Replicators and post it on the blockchain via [DataModificationTransaction](#datamodificationtransaction). The Owner has the possibility to cancel the modification with [DataModificationCancelTransaction](#datamodificationcanceltransaction). In this case the Owner still has to pay the Replicators for the modification.
+
+To close the Drive, the Owner should send a [DriveClosureTransaction](#driveclosuretransaction). Unspent funds will be returned to the Owner.
+
+### Replicator
+
+To provide the Storage, the Replicator should release [ReplicatorOnboardingTransaction](#replicatoronboardingtransaction) and prove its space with collateral. The real free space on the Replicator’s hard Drive MUST be at least twice larger than the mentioned in the [ReplicatorOnboardingTransaction](#replicatoronboardingtransaction) to provide the possibility of rollbacks.
```
Comment (wernling-proximax) [inline docs/built-in-features/storage.md]:
```suggestion
To provide the Storage, the Replicator should release [ReplicatorOnboardingTransaction](#replicatoronboardingtransaction) and prove its space with collateral. The real free space on the Replicator’s hard drive MUST be at least twice larger than mentioned in [ReplicatorOnboardingTransaction](#replicatoronboardingtransaction) to provide the possibility of rollbacks.
```

File: docs/built-in-features/storage.md [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+
+With a special algorithm, the Replicator will be automatically attached to a Drive.
+
+The replicator leaves the Drive if it was closed or the next `Billing Period` has not been paid. Also, the replicator could be excluded from the Drive based on the results of the verification.
+
+To offboard the Replicator should release the [ReplicatorOffBoardingTransaction](#replicatoroffboardingtransaction). The Replicator receives the payments for the work at the end of the `Billing Period`, and returns the deposit for the Drive, but losses part of its Streaming Deposit. This amount compensates for further data uploading during substitution.
+
+### Content Consumer
+
+To be able to download data from a Drive, the Content Consumer should open a Download Channel on this Drive and prepay some amount via [DownloadTransaction](#downloadtransaction). Any node that can issue this transaction indicated the list of Public keys (Content Consumers) that can download files within this channel. If this node wants to grant free downloading to any interested node, it records zero-key in this field. Once per `Download Billing Period (24h)`, some amount of locked money will be paid to the Replicators.
```
Comment (wernling-proximax) [inline docs/built-in-features/storage.md]:
```suggestion
To download data from a Drive, the Content Consumer should open a Download Channel on this Drive and prepay some amount via [DownloadTransaction](#downloadtransaction). Nodes issuing this transaction should contain a list of Public keys (Content Consumers) that can download files within this channel. If a node wants to grant free downloading to any interested node, it should record zero-key in this field. Once per `Download Billing Period (24h)`, some amount of locked money will be paid to the Replicators.
```

File: docs/built-in-features/storage.md [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+**Entity type**: 
+
+**Inlines**:
+
+- [Transaction](../protocol/transaction.md#transaction) or [EmbeddedTransaction](../protocol/transaction.md#embeddedtransaction)
+
+| **Property**    | **Type** | **Description**                                               |
+| --------------- | -------- | ------------------------------------------------------------- |
+| DriveKey        | 32 bytes | Public key of the drive                                       |
+| DownloadDataCdi | 32 bytes | Identifier of the transaction that initiated the modification |
```
Comment (wernling-proximax) [inline docs/built-in-features/storage.md]:
```suggestion
| DataModificationId | 32 bytes | Identifier of the transaction that initiated the modification |
```

File: docs/built-in-features/storage.md [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+
+**Entity type**: 
+
+**Inlines**:
+
+- [Transaction](../protocol/transaction.md#transaction) or [EmbeddedTransaction](../protocol/transaction.md#embeddedtransaction)
+
+| **Property**      | **Type** | **Description**                        |
+| ----------------- | -------- | -------------------------------------- |
+| DownloadChannelId | 32 bytes | The identifier of the download channel |
```
Comment (wernling-proximax) [inline docs/built-in-features/storage.md:234]:
```suggestion
| DownloadChannelId | 32 bytes | The identifier of the download channel             |
| FeedbackFeeAmount | uint64   | Amount of XPXs to transfer to the download channel |
```

File: docs/built-in-features/storage.md [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+
+**Inlines**:
+
+- [Transaction](../protocol/transaction.md#transaction) or [EmbeddedTransaction](../protocol/transaction.md#embeddedtransaction)
+
+| **Property**              | **Type**                                    | **Description**                                   |
+| ------------------------- | ------------------------------------------- | ------------------------------------------------- |
+| DriveKey                  | 32 bytes                                    | Public key of the drive                           |
+| VerificationTrigger       | 32 bytes                                    | The hash of block that initiated the Verification |
+| ProversCount              | uint16                                      | Number of Provers                                 |
```
Comment (wernling-proximax) [inline docs/built-in-features/storage.md]: Are the properties mentioned here meant to match with this?
https://github.com/proximax-storage/cpp-xpx-chain/blob/feature/storage-v3/plugins/txes/storage/src/model/EndDriveVerificationTransaction.h

File: docs/protocol/transaction.md [hunk 7]
```diff
@@ -41,20 +41,36 @@ There are different types of transactions. For example, you can transfer [mosaic
...
 |- | [Deploy Transaction]() | Deploying super contract.|
 |- | [Execute Transaction]() | Start contract executing.|
 |- | [Start Operation Transaction]() | Starting executing operation.|
 |- | [End Operation Transaction]() | Ending executing operation.|
 |- | [Operation Identify Transaction]() | Identifying execute operation.|
+|**Storage**|||
+|0x4662| [Replicator Onboarding Transaction](../guides/storage/replicator-onboarding) | Replicator Onboarding|
+|0x4162| [Prepare BcDrive Transaction](../guides/storage/prepare-bc-drive) | Prepare a New Drive|
+|0x4262| [Data Modification Transaction](../guides/storage/data-modification) | New Data Modification|
+|0x4462| [Data Modification Approval Transaction]() | Data Modification Approval. Inner transaction, sends Only by Replicators |
```
Comment (wernling-proximax) [inline docs/protocol/transaction.md]: Will there be a webpage link added for each approval-type transaction in the future?
```suggestion
|0x4462| [Data Modification Approval Transaction]() | Data Modification Approval. Inner transaction, sent by Replicators only |
```

**Pull request merged**